lib/goth/http_client/finch.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Goth.HTTPClient.Finch do
```

Comment:
please remove this module as we want to get away from the `http_client: {mod, opts}` API. Since Finch integration is a new addition, nobody used it yet so there's no code we'd break.